### PR TITLE
Issue610 cdl extra basic blocks

### PR DESCRIPTION
**Pull request by JayHuLBL**
Additional classes were added:
- Additional continuous block `CDL.Continuous.RampLimiter`;
- Validation package in `CDL.Conversions`;
- Example package in `CDL.Discrete`;
- Validation package in `CDL.Integers`;
- Additional logical block `CDL.Logical.Latch`, `CDL.Logical.Toggle`;
- Validation package in `CDL.Logical`;
- Additional source block `CDL.Sources.DutyCycle`;
- Validation package in `CDL.Sources`

### CODE REVIEW

File: Buildings/Experimental/OpenBuildingControl/CDL/Continuous/Truncation.mo [hunk 1]
```diff
@@ -6,7 +6,7 @@ block Truncation
     annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
         iconTransformation(extent={{-140,-20},{-100,20}})));
 
-  Interfaces.RealOutput y "Connector of Real output signal"
+  Interfaces.IntegerOutput y "Connector of Real output signal"
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Continuous/Truncation.mo]: Comment should be "...Integer output signal"

File: Buildings/Experimental/OpenBuildingControl/CDL/Continuous/Validation/RampLimiter.mo [hunk 2]
```diff
@@ -0,0 +1,47 @@
+within Buildings.Experimental.OpenBuildingControl.CDL.Continuous.Validation;
+model RampLimiter "Validation model for the RampLimiter block"
+extends Modelica.Icons.Example;
+
+  Buildings.Experimental.OpenBuildingControl.CDL.Continuous.RampLimiter ramplimiter1(
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Continuous/Validation/RampLimiter.mo]: Use `ramLim` (or `rampLimiter`)

File: Buildings/Experimental/OpenBuildingControl/CDL/Conversions/Validation/BooleanToInteger.mo [hunk 3]
```diff
@@ -0,0 +1,39 @@
+within Buildings.Experimental.OpenBuildingControl.CDL.Conversions.Validation;
+model BooleanToInteger
+  "Validation model for the BooleanToInteger block"
+extends Modelica.Icons.Example;
+
+  Buildings.Experimental.OpenBuildingControl.CDL.Conversions.BooleanToInteger BooToIn1
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Conversions/Validation/BooleanToInteger.mo]: use `booToInt`

File: Buildings/Experimental/OpenBuildingControl/CDL/Continuous/RampLimiter.mo [hunk 4]
```diff
@@ -0,0 +1,93 @@
+within Buildings.Experimental.OpenBuildingControl.CDL.Continuous;
+block RampLimiter "Limit the increase or decrease rate of input"
+
+  parameter Real Increase "Increase amount of the input";
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Continuous/RampLimiter.mo]: use `increase` and `decrease`

File: Buildings/Experimental/OpenBuildingControl/CDL/Continuous/RampLimiter.mo [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  parameter Real Decrease "Decrease amount of the input";
+
+  parameter Modelica.SIunits.Time  IncDT(min=100*Constants.eps) = 1.0
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Continuous/RampLimiter.mo]: use `incDelTim`

File: Buildings/Experimental/OpenBuildingControl/CDL/Continuous/RampLimiter.mo [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  else
+    y = u;
+  end if;
+   annotation (
+Documentation(info="<html>
+<p>
+The block limits the increase/decrease rate of its input signal in the range of <code>[Falling, Rising]</code>, where:
+</p>
+<pre>
+    <code>Falling = -Decrease/DecDT</code>; <code>Rising = Increase/IncDT</code>;
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Continuous/RampLimiter.mo]: Don't use <code> instead the <pre> section, as <pre> already typesets it as code.

File: Buildings/Experimental/OpenBuildingControl/CDL/Conversions/Validation/IntegerToReal.mo [hunk 7]
```diff
@@ -0,0 +1,41 @@
+within Buildings.Experimental.OpenBuildingControl.CDL.Conversions.Validation;
+model IntegerToReal "Validation model for the IntegerToReal block"
+extends Modelica.Icons.Example;
+
+  Buildings.Experimental.OpenBuildingControl.CDL.Conversions.IntegerToReal InToReal1
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Conversions/Validation/IntegerToReal.mo]: use `intToRea` (or `intToReal`)

File: Buildings/Experimental/OpenBuildingControl/CDL/Logical/Validation/FallingEdge.mo [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  Buildings.Experimental.OpenBuildingControl.CDL.Sources.DutyCycle DutyCyc1(
+    CycleOn = true,
+    period = 1.0)
+    "Block that output cyclc on and off"
+    annotation (Placement(transformation(extent={{-36,-8},{-16,12}})));
+
+  Buildings.Experimental.OpenBuildingControl.CDL.Continuous.Constant cons1(k=50) "Constant as source term"
+    annotation (Placement(transformation(extent={{-76,-8},{-56,12}})));
+
+  Buildings.Experimental.OpenBuildingControl.CDL.Logical.FallingEdge Fedge1
```
Comment (mwetter) [inline Buildings/Experimental/OpenBuildingControl/CDL/Logical/Validation/FallingEdge.mo]: use `falEdg` (e.g., camel case notation)

**Comment by milicag** [comment]
@JayHuLBL @mwetter I intend to introduce ```Or3```,  and ```ZoneState``` type with enumeration in the economizer sequence and its atomic sequences once we have them in the basic blocks. 

G36 explains ```Zone States``` on pg. 43. An another type we could add at some point is ```AHUMode```, see pg. 103. My idea is that for AHU Mode, it would be usefull to enumerate with integers starting at 1 as listed in G36 (Occupied Mode is 1), whereas for the Zone State we could assign 1 to Heating, -1 to Cooling and 0 to Deadband. ```AHU Mode``` is being used to control fans, airflow requirement, economizer, etc. I'm happy to implement this if you find it useful.  We could also aks Brent to which extent these string type multiple choice variables resemble what is currently implemented in control systems.

**Comment by mwetter** [comment]
@milicag Enumeration should start at 1, and be increasing. This is also how Modelica defines enumerations.

**Comment by milicag** [comment]
@mwetter What is a good option if we'd like to use ```Cooling``` Zone State as an input signal. For example, I'd like to add -2 degrees to the setpoint it Zone State = ```Cooling```. I assume we could use string comparrison.

**Comment by mwetter** [comment]
@milicag If cooling is an enumeration, you can check for equality with this enumeration. String comparison would be very expensive.

**Comment by JayHuLBL** [comment]
@mwetter I did following changes since your last review:
-- Updated: variable names, add experiment notation, add `P/PI/PD/noLimPID` validation;
-- Updated `CDL.Psychrometrics` package;
-- Added blocks (and Validation): `BooleanToReal/And3/Or3`
Would you please review the changes? Thanks.

**Comment by milicag** [comment]
Note that there already is a ```Type``` named ```ZoneState```, so if we wish to implement the G36 defined "Zone State", we'll need to rename one of the two. At my current level of understanding of the guidline, it looks like their sequences allow for modularity in approaching building controls, so our first implementation should try to relly on the standard as much as possible/reasonable.

**Pull request merged**